dotcom-rendering/src/web/components/SignInGate/displayRule.ts

```diff
@@ -69,38 +68,8 @@ export const isValidTag = (tags: TagType[]): boolean => {
 };
 
 export const hasRequiredConsents = (): Promise<boolean> => {
-	const hasConsentedToAll = (state: ConsentState) => {
-		const consentFlags = state.tcfv2?.consents
-			? Object.values(state.tcfv2.consents)
-			: [];
-		const vendorConsentFlags = state.tcfv2?.vendorConsents
-			? Object.values(state.tcfv2.vendorConsents)
-			: [];
-		const isEmpty =
-			consentFlags.length === 0 || vendorConsentFlags.length === 0;
-
-		return (
-			!isEmpty && [...consentFlags, ...vendorConsentFlags].every(Boolean)
-		);
-	};
-
 	return new Promise((resolve) => {
```

Comment:
Perhaps a candidate for [onConsent](https://github.com/guardian/consent-management-platform#onconsent) ?